lib/libc/picolibc/CMakeLists.txt

```
@@ -11,7 +11,10 @@ zephyr_library_sources(
   stdio.c
   )
 
-zephyr_library_compile_options(-fno-lto)
+zephyr_check_compiler_flag(C -fno-lto check_no_lto_result)
```

the current code on main seems to be a bad use of hard-coding `-fno-lto` as the compiler-flags has a property for exactly the reason of supporting older gcc versions / different compiler flags for this option:
https://github.com/zephyrproject-rtos/zephyr/blob/8b987d9e0293b1616ae738cdbee0bac33b71c271/cmake/compiler/gcc/compiler_flags.cmake#L25-L28

```suggestion
zephyr_library_compile_options($<TARGET_PROPERTY:compiler,prohibit_lto>)
```

Feel free to update `compiler_flags.cmake` to use `zephyr_check_compiler_flag()` instead of base the property on the compiler version.

---

That is certainly an improvement. Thanks for the suggestion.